server/web/webserver_test.go

```diff
@@ -0,0 +1,89 @@
+package web
+
+import (
+	"net/url"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+func Test_NewServer(t *testing.T) {
+	tests := []struct {
+		name       string
+		rootPath   string
+		serverRoot string
+		ssl        bool
+		port       int
+		localOnly  bool
+		logger     *mlog.Logger
+	}{
+		{
+			name:       "should return Server with given properties",
+			rootPath:   "./test/path/to/root",
+			serverRoot: "https://some-fake-server.com/fake-url",
+			ssl:        false,
+			port:       9999, // fake port number
+			localOnly:  false,
+			logger:     &mlog.Logger{},
+		},
+		{
+			name:       "should return local server with given properties",
+			rootPath:   "./test/path/to/root",
+			serverRoot: "https://some-fake-server.com/fake-url",
+			ssl:        false,
+			port:       3000, // fake port number
+			localOnly:  true,
+			logger:     &mlog.Logger{},
+		},
+		{
+			name:       "should match Server properties when ssl true",
+			rootPath:   "./test/path/to/root",
+			serverRoot: "https://some-fake-server.com/fake-url",
+			ssl:        true,
+			port:       8000, // fake port number
+			localOnly:  false,
+			logger:     &mlog.Logger{},
+		},
+		{
+			name:       "should return local server with given properties",
+			rootPath:   "./test/path/to/root",
+			serverRoot: "https://localhost:8080/fake-url",
+			ssl:        true,
+			port:       9999, // fake port number
+			localOnly:  true,
+			logger:     &mlog.Logger{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			baseURL := ""
+			url, err := url.Parse(test.serverRoot)
+			if err != nil {
+				test.logger.Error("Invalid ServerRoot setting", mlog.Err(err))
+			}
+			baseURL = url.Path
+
+			ws := NewServer(test.rootPath, test.serverRoot, test.port, test.ssl, test.localOnly, test.logger)
+
+			require.NotNilf(t, ws, "The webserver object is nil!")
```

Comment:
nit: the `NotNilf` can be `NotNil` because you are not using formatting there. The same for `Equalf` later. Anyway, this is just a tiny thing that doesn't matter too much.